CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+Instead, you should use `git remote` to add the fork as a new remote for the original repo,
+`$GOPATH/src/github.com/cosmos/ethermint `, and do all the work there.
+
+For instance, to create a fork and work on a branch of it, One would:
+
+  * Create the fork on github, using the fork button.
+  * Go to the original repo checked out locally (i.e. `$GOPATH/src/github.com/cosmos/ethermint`)
+  * `git remote rename origin upstream`
+  * `git remote add origin git@github.com:<username>/ethermint.git`
+
+Now `origin` refers to my fork and `upstream` refers to the Cosmos-SDK version.
+So I can `git push -u origin <branch>` to update my fork, and make pull requests to Cosmos-SDK from there.
+Of course, replace `<username>` with your git handle.
+
+To pull in updates from the origin repo, run
+
+    * `git fetch upstream`
+    * `git rebase upstream/develop` (or whatever branch you want)
+
+Do not make Pull Requests to `master`, they will not be considered.
+
+## Dependencies
+
+We use [`dep`](https://github.com/golang/dep) to manage dependencies.
+
+That said, the master branch of every Cosmos repository should just build
+with `go get`, which means they should be kept up-to-date with their
+dependencies so we can get away with telling people they can just `go get` our
+software.
+
+Since some dependencies are not under our control, a third party may break our
+build, in which case we can fall back on `dep ensure` (or `make
+get_vendor_deps`). Even for dependencies under our control, `dep` helps us to
+keep multiple repos in sync as they evolve. Anything with an executable, such
+as apps, tools, and the core, should use `dep`.
+
+Run `dep status` to get a list of vendor dependencies that may not be
+up-to-date.
+
+## Testing
+
+All repos should be hooked up to [CircleCI](https://circleci.com/).
+
+If they have `.go` files in the root directory, they will be automatically
+tested by circle using `go test -v -race ./...`. If not, they will need a
+`circle.yml`. Ideally, every repo has a `Makefile` that defines `make test` and
+includes its continuous integration status using a badge in the `README.md`.
+
+## Branching Model and Release
+
+User-facing repos should adhere to the branching model: http://nvie.com/posts/a-successful-git-branching-model/.
+That is, these repos should be well versioned, and any merge to master requires a version bump and tagged release.
+
+Libraries need not follow the model strictly, but would be wise to.
+
+The SDK utilizes [semantic versioning](https://semver.org/).
+
+### Development Procedure:
+- the latest state of development is on `develop`
+- `develop` must never fail `make test`
+- `develop` should not fail `make test_lint`
+- no --force onto `develop` (except when reverting a broken commit, which should seldom happen)
+- create a development branch either on github.com/cosmos/cosmos-sdk, or your fork (using `git remote add origin`)
+- before submitting a pull request, begin `git rebase` on top of `develop`
```

Comment:
I'd add a note here suggesting it'd be wise to squash commits before rebasing.

Reply:
Done, also added link to procedure